crates/syntax/src/ast/node_ext.rs

```diff
@@ -53,8 +53,21 @@ impl ast::Attr {
     pub fn as_simple_key_value(&self) -> Option<(SmolStr, SmolStr)> {
         let lit = self.literal()?;
         let key = self.simple_name()?;
-        // FIXME: escape? raw string?
-        let value = lit.syntax().first_token()?.text().trim_matches('"').into();
```

Comment:
I think we already have "correct" way to do this implemented: https://github.com/rust-analyzer/rust-analyzer/blob/ea25ae614b21237c4a536304da875bdc29f0c65a/crates/syntax/src/ast/token_ext.rs#L149

We can just `ast::String::cast` our way to victory here, but, ideally, we should improve the API surrounding string literals somewhat: #6308

Reply:
Thanks for pointing this out! I've updated this to cast to `ast::String`/`ast::RawString` then use the `value` method. 

I did take a look at #6308 but I think that would take a bit more time than I have right now. But I understand if you'd rather handle this after that, so feel free to close this PR if so.

Thanks for the review!

Reply:
:+1: 

bors r+